art_tree.go

```diff
@@ -59,7 +59,7 @@ func (t *ArtTree) searchHelper(current *ArtNode, key []byte, depth int) interfac
 // Inserts the passed in value that is indexed by the passed in key into the ArtTree.
 func (t *ArtTree) Insert(key []byte, value interface{}) {
 	key = ensureNullTerminatedKey(key)
-	t.insertHelper(t.root, &t.root, key, value, 0)
+	t.insertHelper(t.root, &t.root, key, key, value, 0)
```

Comment:
Don't we want to pass in the non null-terminated key here?

I think we might want to save a temporary variable and pass it into the helper:

```
func(t *ArtTree) Insert(key []byte, value interface{}) {
  completeKey := key
  key = ensureNullTerminatedKey(key)
  t.insertHelper(t.root, &t.root, key, completeKey, value, 0)
}
```